Maybe using belongsTo or hasMany could be helpful in optimizing this. I know Mostafa mentioned he has code regarding that in his branch. We can revisit this and try optimizing it later if we have time left.

For now, resolve these eslint issues and go ahead with the merge:
CapitalOneFraudDetection/src/components/TransactionCard.tsx
  40:12  error  Trailing spaces not allowed                    @stylistic/no-trailing-spaces
  45:1   error  Expected indentation of 8 spaces but found 12  @stylistic/indent

Thanks!

--- a/src/components/TransactionCard.tsx
+++ b/src/components/TransactionCard.tsx
@@ -37,12 +37,12 @@ const TransactionCard: React.FC<TransactionProps> = ({ transaction }) => {
         }
         const { data: accountData } = await client.models.Account.get({
             id: transaction.accountID,
-        });   
+        });
         if (!accountData || !accountData.accountName) {
             console.error("Account name is missing or undefined.");
             return;
         }
-            navigate("/transaction-details", {
+        navigate("/transaction-details", {
             state: {
                 transaction,
                 accountName: accountData.accountName,

--- a/src/pages/NewAccountPage.tsx
+++ b/src/pages/NewAccountPage.tsx
@@ -46,8 +46,7 @@ export default function NewAccountPage() {
             setIsLoading(false);
             setShowSuccess(true);
             reset();
-        } catch (error) {
-            console.error("Error creating account:", error);
+        } catch {
             setIsLoading(false);
             setShowFailure(true);
             reset();

--- a/src/pages/NewTransactionPage.tsx
+++ b/src/pages/NewTransactionPage.tsx
@@ -135,8 +135,7 @@ export default function NewTransactionPage() {
             setIsLoading(false);
             setShowSuccess(true);
             reset();
-        } catch (error) {
-            console.error("Error with transaction:", error);
+        } catch {
             setIsLoading(false);
             setShowFailure(true);
             reset();

